WcaOnRails/app/controllers/api/v0/api_controller.rb

```diff
@@ -9,9 +10,17 @@ def bad_api_parameter(e)
 
   DEFAULT_API_RESULT_LIMIT = 20
 
+  def current_resource_owner
+    if doorkeeper_token
+      User.find_by_id(doorkeeper_token.resource_owner_id).tap do |user|
+        user.doorkeeper_token = doorkeeper_token
```

Comment:
Doesn't this suffer from the same problem as the original? If we don't find the user, then we're going to blow up trying to do `user.doorkeeper_token = ...`.

Also, am I the only one who feels weird about going to the database *knowing* we're not supposed to find something? With this code, we'll end up calling `User.find_by_id(nil)`, hoping/expecting it will return `nil`.